common4j/src/main/com/microsoft/identity/common/java/broker/CommonTenantInfoProvider.kt

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.broker
+import com.microsoft.identity.common.java.interfaces.ITenantInfoProvider
+import com.microsoft.identity.common.java.logging.Logger
+
+/**
+ * Consumer of commons needs to implement [ITenantInfoProvider] interface
+ * and set it using CommonTenantInfoProvider.initializeCommonTenantInfoProvider(@NonNull tenantInfoProvider: ITenantInfoProvider)
+ * to provide tenantInfo to common module.
+ */
+object CommonTenantInfoProvider : ITenantInfoProvider{
+    private val TAG = CommonTenantInfoProvider::class.java.simpleName
+    private var mTenantInfoProvider: ITenantInfoProvider? = null
+
+    // Note : This method should only be invoked by broker module.
```

Comment:
How do we ensure that?

Reply:
Also what would go wrong if someone else invokes it?

Reply:
It just means that right now we only support getting tenant related information at this stage using broker account storage. May be in the future, if we can provide tenant information from common module itself, we can invoke it. Since the comment is confusing, i would rather remove it.